docs/latex/Inhalt/03_Inhalt/03_clean_architecture.tex

```diff
@@ -0,0 +1,61 @@
+
+\section{Schicht 0: Plugins und Main}
+
+\subsection{Plugin CLI}
+Dieses Plugin, welches im Module \code{0-quickie-plugin-cli} implementiert wurde und die Benutzeroberfläche der Software bildet, enthält lediglich die Klasse \code{CommandLineUI}. Diese implementiert das Interface \code{UI} aus der Adapterschicht. Abgesehen von dieser besitzt dieses Plugin nur eine weitere Dependency: Die Library Apache Commons CLI, welche zur leichteren Implementierung der CLI verwendet wird. Das Plugin kommuniziert lediglich über die Eventfunktionalität der Adapterschicht mit dem Kern der Software, ist damit nur sehr lose an den selben gekoppelt und kann leicht ausgetauscht werden.
+
+\subsection{Plugin JPA}
+Dieses Plugin wurde im Module \code{0-quickie-plugin-jpa} realisiert die Persistenz der Software mit Hilfe der \ac{JPA} und deren Implementierung Hibernate. Auch dieses Plugin besitzt nur eine Dependency auf die Adapterschicht, da hier die dort im Package \code{persistence} definierten Interfaces für Factories und die zu persistierenden Objekte Recipe, Ingredient und Profile implementiert werden. Letztere enthalten außerdem die \ac{JPA}-Annotationen. Ferner enthält das Plugin die Klassen \code{JPARecipeRepository} und \code{JPAProfileRepository}, welche von den beiden abstrakten Repository-Klassen aus der Adapterschicht erben. Schließlich enthält das Plugin noch den \code{PersistenceMananger}, einen Singleton, welcher den \code{EntityManager} von \code{JPA} für die Repositories liefert.
+
+\subsection{Plugin Scraper}
+In diesem Plugin wird die Extraktion der Rezepte aus den entsprechenden Webseiten im Module \code{0-quickie-plugin-scraper} implementiert. Hierzu werden die Webseiten zunächst durch den \code{CachedDownloader} heruntergeladen. Für das Parsen der Webseiten wird der \code{HensslerScraper} verwendet, welcher die Rezeptdaten aus dem HTML-Text extrahiert. Diese werden dann durch die \code{HensslerDatasource}, welche das in der Application Layer definierte Interface \code{Datasource} implementiert, für die Application Layer zugänglich gemacht. Die Daten durchlaufen damit nicht die Adapterschicht, da der Scraper die Daten problemlos direkt in dem von der Application Layer benötigten Format liefern kann.
+
+\subsection{Main}
+Das Module \code{0-quickie-main} enthält lediglich die Main-Klasse der Applikation, welche alle nötigen Klassen instanziiert und ggf. injected. Zuletzt wird hier auch die Methode \code{run} des Controllers aus der Adapterschicht ausgeführt und somit die Verarbeitung von Events gestartet. Dieses Modul besitzt folglich Abhängigkeiten auf alle anderen Module der Software und befindet sich daher in der äußersten Schicht.
```

Comment:
"alle anderen Module" => "alle anderen Modules"

Reply:
"in der äußersten Schicht" => Begründung, wieso wir dafür keine Extra-Schicht haben?

Reply:
Warum sollten wir dafür eine Extra-Schicht haben? Die Schichten sind ja durch die Clean Architecture vorgegeben und weiter raus als Schicht 0 geht halt schlecht oder?